src/ProfileView.jl

```diff
@@ -484,12 +481,13 @@ end
 
 @guarded function save_as_cb(::Ptr, profdata::Tuple)
     c, data, lidict, g = profdata
-    selection = save_dialog("Save profile data as *.jlprof file", toplevel(c), ("*.jlprof",))
-    isempty(selection) && return nothing
-    if data === nothing && lidict === nothing
-        return _save(selection, g)
+    save_dialog("Save profile data as *.jlprof file", toplevel(c), ("*.jlprof",)) do selection
+        isempty(selection) && return nothing
+        if data === nothing && lidict === nothing
+            _save(selection, g)
+        end
```

Comment:
```suggestion
        if data === nothing && lidict === nothing
            _save(selection, g)
        else
            _save(selection, data, lidict)
        end
```
This fixes the saving functionality.